test/e2e/specs/ExistingUserCreditCheckout.spec.js

```diff
@@ -0,0 +1,51 @@
+describe('Existing User Checkout with Kiva Credit', () => {
+	it('Logs in user with credit, baskets a loan, proceed to checkout, and purchases using kiva credit', () => {
+		let beginBalance;
+		let newBalance;
+		// get the test user id from cypress.env.json and set the cookie variable
+		const lenderID = Cypress.env().userInfo.kivaCreditLender.userID;
+		const loginCookie = `${lenderID}:recent/active/mfa`;
+
+		// Mock login our test lender
+		cy.setCookie('kvfa', loginCookie, 'domain=.kiva.org');
+		// go to portfolio and get current credit for later calculations
+		cy.visit('/portfolio');
+		cy.get('.strong .kiva-green').invoke('text').then(fullText => {
+			const number = fullText.match(/[0-9,.]+/);
+			beginBalance = parseFloat(number); // text to float
+		});
+		// Go to category page for women
+		cy.visit('/lend-by-category/women');
+		// // Wait for the lend or lend again button to exist, then click it
```

Comment:
Rather than using an explicit timeout for these commands, I would recommend intercepting the GraphQL requests before the `cy.setCookie` and `cy.visit` commands, and waiting for all those requests to finish each time:

```js
// Spy on GraphQL requests
cy.intercept('POST', '**/graphql*').as('graphqlRequest');
// Mock login our test lender
cy.setCookie('kvfa', loginCookie, 'domain=.kiva.org');
// go to portfolio and get current credit for later calculations
cy.visit('/portfolio');
cy.get('.strong .kiva-green').invoke('text').then(fullText => {
	const number = fullText.match(/[0-9,.]+/);
	beginBalance = parseFloat(number); // text to floats
});
// Go to category page for women
cy.visit('/lend-by-category/women');
// Wait for the lend or lend again button to exist, then click it
cy.get('@graphqlRequest.all');
cy.findAllByText('Lend now').first().click();
// wait for the basket add lightbox
cy.get('@graphqlRequest.all');
cy.get('.kv-lightbox').should('be.visible');
// Wait for the checkout button to exist in the lightbox, then click it
cy.get('.button-checkout').click();
// Wait for basket to load and check for complete order button
cy.get('@graphqlRequest.all');
cy.findByText('Basket');
cy.findByText('Complete order').click();
// wait for order confirmation page to load
cy.get('@graphqlRequest.all');
cy.findByText('Thank you!');
```

Reply:
I like this approach to understanding when you are clear to take the next action.

On another note, I realized there's another exit out of the UI app to portfolio. We should be able to get the balance from the Header and not leave the Ui app.

Reply:
@mcstover the reason why I did not use the amount from the header is because it displays a rounded number.  We need to have the full amount including cents and the portfolio page has that.

Reply:
@emuvente thanks for the recommendation, I have updated the test to use cy.intercept.